src/arrayMethodReduce.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const callbackFunction = callback;
+    const accumulator = arguments.length < 2
+      ? this[0]
+      : initialValue;
+    let reducedValue = accumulator;
+
+    const indexStart = arguments.length < 2
+      ? 1
+      : 0;
```

Comment:
you check for the same thing twice, in this case it's better to use a regular if statement